packages/reaction-accounts/server/methods/accounts.js

```diff
@@ -127,20 +127,33 @@ Meteor.methods({
     return false;
   },
 
-  /*
-   * add new addresses to an account
+  /**
+   * accounts/addressBookAdd
+   * @description add new addresses to an account
+   * @param {Object} doc - address
+   * @param {String} accountId - `account._id` which is need to be updated
+   * @return {Object} with keys `numberAffected` and `insertedId` if doc was
+   * inserted
    */
   "accounts/addressBookAdd": function (doc, accountId) {
     check(doc, ReactionCore.Schemas.Address);
     check(accountId, String);
+    if (accountId !== Meteor.userId() && !ReactionCore.hasAdminAccess()) {
```

Comment:
Shouldn't this be:

`if (accountId !== Meteor.userId() || ReactionCore.hasAdminAccess()) `?